app/components/export-tool/component.js

```diff
@@ -1,138 +1,184 @@
-        'TSV (for ISP)': 'TSV',
+        CSV: 'CSV',
+        'CSV (for ISP)': 'CSV'
     },
 
-    processedData: Ember.computed('data', 'dataFormat', function() {
-        var data = this.get('data') || [];
+    /**
+     * @property {null|string} The processed data
+     */
+    processedData: Ember.computed('data', 'dataFormat', 'mappingFunction', function() {
+        let data = this.get('data') || [];
+
         if (data.toArray) {
             data = data.toArray();
         }
-        let dataArray = data.map(serializeItem);
 
-        var dataFormat = this.get('dataFormat');
-        var mappingFunction = this.get('mappingFunction');
+        const dataArray = data.map(serializeItem);
+        const dataFormat = this.get('dataFormat');
+        const mappingFunction = this.get('mappingFunction');
 
         if (Ember.isPresent(dataArray)) {
-            let mapped = dataArray.map(mappingFunction);
+            const mapped = mappingFunction ? dataArray.map(mappingFunction) : dataArray;
             return this.convertToFormat(mapped, dataFormat);
-        } else {
-            return null;
         }
+
+        return null;
     }),
 
+    /**
+     * Converts an array to a JSON string
+     *
+     * @param {Array<Object>} dataArray An array of objects
+     * @private
+     * @returns {string}
+     */
     _convertToJSON(dataArray) {
         return JSON.stringify(dataArray, null, 4);
     },
-    _convertToTSV(dataArray) {
-        // Flatten the dictionary keys for readable column headers
-        let squashed = dataArray.map((item => squash(item)));
-
-        var fields = Object.keys(squashed[0]);
-        var tsv = [fields.join('\t')];
 
-        squashed.forEach((item) => {
-            var line = [];
-            fields.forEach(function(field) {
-                line.push(JSON.stringify(item[field]));
-            });
-            tsv.push(line.join('\t'));
-        });
-        tsv = tsv.join('\r\n');
-        return tsv;
+    /**
+     * Converts an array to a standard CSV file
+     *
+     * @param {Array<Object>} dataArray The rows of the CSV
+     * @private
+     * @returns {string}
+     */
+    _convertToCSV(dataArray) {
+        const data = dataArray.map(squash);
+        const fields = uniqueFields(data);
+        return writeCSV(data, fields);
     },
+
+    /**
+     * Converts an array to a standard CSV file
+     *
+     * @param {Array<Object>} dataArray The rows of the CSV
+     * @private
+     * @returns {string}
+     */
     _convertToISP(dataArray) {
-        // ISP-specific TSV file format
+        // ISP-specific CSV file format
 
         // First custom field mapping...
 
-        // Then serialize to TSV
+        // Then serialize to CSV
         dataArray = dataArray.map((record) => {
-            let newRecord = {};
-            for (let frameId of Object.keys(record.expData)) {
-                newRecord[frameId] = record.expData[frameId].responses || {};
+            console.log(record);
```

Comment:
I understand this is a work in progress; just noting console.log statement for myself to make sure it's removed before merge.